### PR TITLE
Feature: Settings to enable telemetry for extension

### PR DESCRIPTION
**Pull request by hassan-pullflow**
#### Summary of Changes 

- added telemetry settings in `package.json`
- initialising telemetry in `initialize.ts`
-  added `isTelemetryEnabled` flag in `CacheObject`

### Snap-Shots 

Screenshot 2024-01-08 at 5 27 11 PM

### PR-Specific-Tests

- Check by default telemetry is sent to grafana > Disable telemetry from vscode settings > Telemetry is not being sent to grafana.

### CODE REVIEW

**Review by srzainab — APPROVED** [review]

File: src/utils/trace.ts [hunk 1]
```diff
@@ -10,13 +11,47 @@ import { ExtensionContext, extensions } from 'vscode'
...
+import { log } from './logger'
 
 const extensionInfo = extensions.getExtension('Pullflow.pullflow')?.packageJSON
 
-export class Trace {
+type FakeTracer = {}
+type FakeBasicTracerProvider = {}
+type FakeAttribute = {}
+
+// Null Object Design pattern
```
Comment (amna-pullflow) [inline src/utils/trace.ts:22]: do we need this comment here?

File: src/utils/trace.ts [hunk 2]
```diff
@@ -10,13 +11,47 @@ import { ExtensionContext, extensions } from 'vscode'
...
+
+  constructor(_context: ExtensionContext) {
+    this.provider = {}
+    this.tracer = {}
+    this.defaultAttributes = {}
+  }
+
+  dispose(): void {}
+  start({ name, attributes }: { name: string; attributes?: TraceAttributes }) {
+    console.log({ name, attributes })
```
Comment (amna-pullflow) [inline src/utils/trace.ts:44]: why are we doing console log here?

**Review by amna-pullflow — CHANGES_REQUESTED** [review]

File: src/utils/trace.ts (hunk 2 above)
Comment (srzainab) [reply src/utils/trace.ts:44]: because i need to use the attributes and name somehow otherwise it gives unused values errors.

**Review by srzainab — APPROVED** [review]

File: src/utils/trace.ts (hunk 2 above)
Comment (amna-pullflow) [reply src/utils/trace.ts:43]: we can do it like one of these ways:
`start({ _name, _attributes }: { name: string; attributes?: TraceAttributes }) { ...
}`

`start({ }: { name: string; attributes?: TraceAttributes }) { ...
}`

what do you think @srzainab?

**Review by amna-pullflow — APPROVED** [review]
left a comment regarding the unused values error, looks good otherwise

**Comment by srzainab** [comment]
@amna-pullflow updated the PR. will merge once the tests pass.

**Pull request merged**